app/models/volunteer.rb

```diff
@@ -71,6 +73,12 @@ class Volunteer < ApplicationRecord
     joins(:assignments).merge(Assignment.active_between(start_date, end_date)).distinct
   }
   scope :without_assignment, (-> { left_outer_joins(:assignments).where(assignments: { id: nil }) })
+  scope :without_active_assignment, (-> { joins(:assignments).merge(Assignment.ended) })
+  scope :not_responsible, (-> { where.not(id: GroupOffer.pluck(:responsible_id)) })
```

Comment:
This will run two queries, you can do it all in one by using `where.not(id: GroupOffer.select(:responsible_id))`

This generates a subquery which looks like `SELECT ... WHERE id IN (SELECT id ...)` (see Rails log or console)